dotnet/src/InternalUtilities/planning/Functions/KernelPluginFunctionMetadata.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+
+#pragma warning disable IDE0130
+
+namespace Microsoft.SemanticKernel;
+
+/// <summary>
+/// Provides read-only metadata for an <see cref="KernelFunction"/>.
+/// </summary>
+public sealed class KernelPluginFunctionMetadata
```

Comment:
Should we consider using `record` instead?

```suggestion
public sealed record KernelPluginFunctionMetadata
```